### PR TITLE
Milestone3/associations

### PR DESCRIPTION
**Pull request by omarsalem7**
## What does this PR do?

This PR is raised to set up associations as in UML diagram.

## Description of the task to be completed

- [x] Create a class `Classroom `
- [x] Create the `has-many/belongs-to` relationship between Classroom and Student.
- [x] Create a class `Book`
- [x] Create a class `Rental`
- [x] Create the `many-to-many (also has-many-through) ` relationship between `Person` and `Book` using the intermediate class `Rental`

## How should it be tested

Clone the repository
```
git clone -b milestone3/associations https://github.com/omarsalem7/OOP-school-library.git
```
Navigate to the folder
```
cd OOP-school-library
```
Open in VS Code 
```
code .
```

#### Thanks for reviewing my PR 😊

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require_relative('./rental')
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (mikethreels) [inline book.rb]: - [x] to make the associations work *Getters and Setters for @rentals should be present.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_reader :person, :book
```
Comment (mikethreels) [inline rental.rb]: - [x] to make the associations work Getters and Setters for @book and @person instance variables should be present

File: person.rb [hunk 3]
```diff
@@ -1,14 +1,16 @@
 require_relative './nameable'
+require_relative('./rental')
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (mikethreels) [inline person.rb]: - [x] to make the associations work Getters and Setters for @rentals should be present.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Omar, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require_relative('./rental')
+class Book
+  attr_accessor :title, :author, :rentals
```
Comment (mateo951) [inline book.rb:3]: Addressed 😎

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :person, :book, :date
```
Comment (mateo951) [inline rental.rb:2]: Addressed 😉

File: person.rb [hunk 6]
```diff
@@ -1,14 +1,16 @@
 require_relative './nameable'
+require_relative('./rental')
 
 class Person < Nameable
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
```
Comment (mateo951) [inline person.rb:5]: Addressed 😬

**Review by mateo951 — APPROVED** [review]
### Hi @omarsalem7, 

Well done making all the necessary changes requested by the previous reviewer 🔥
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
### Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**